File: ambeth-integrity/de.osthus.classbrowser.csharp/CsharpClassbrowser/CsharpClassbrowser/TypeDescription.cs
using System;
using System.Collections.Generic;

namespace CsharpClassbrowser
{
    /// <summary>
    /// Holds the description for types.
    /// </summary>
    public class TypeDescription : CsharpClassbrowser.INamed
    {

        // ============================================================================================
        #region Constructors
        // ============================================================================================

        /// <summary>
        /// Create a new instance.
        /// </summary>
        /// <param name="source">Source; mandatory</param>
        /// <param name="moduleName">Module name; mandatory (but may be empty if the module can't be found)</param>
        /// <param name="namespaceName">Namespace name; optional</param>
        /// <param name="typeName">Short name of the type; mandatory</param>
        /// <param name="fullTypeName">Full name of the type including the namespace; mandatory</param>
        /// <param name="typeType">Type constant; mandatory</param>
        /// <param name="genericTypeParams">Number of generic type parameters</param>
        public TypeDescription(string source, string moduleName, string namespaceName, string typeName, string fullTypeName, string typeType, int genericTypeParams)
        {
            if (string.IsNullOrWhiteSpace(source) || string.IsNullOrWhiteSpace(typeName) || string.IsNullOrWhiteSpace(fullTypeName) ||
                string.IsNullOrWhiteSpace(typeType) || moduleName == null)
            {
                throw new ArgumentNullException("Mandatory type description value missing!");
            }
            this.Source = source;
            this.ModuleName = moduleName;
            this.TypeType = typeType;
            this.NamespaceName = namespaceName;
            this.Name = typeName;
            this.FullTypeName = fullTypeName;
            this.GenericTypeParams = genericTypeParams;
            this.Annotations = new List<String>();
            this.MethodDescriptions = new List<MethodDescription>();
            this.FieldDescriptions = new List<FieldDescription>();
        }

        #endregion

        // ============================================================================================
        #region Properties
        // ============================================================================================

        public string Source { get; internal set; }
        public string ModuleName { get; internal set; }
        public string TypeType { get; internal set; }
        public string NamespaceName { get; internal set; }
        public string Name { get; internal set; }
        public string FullTypeName { get; internal set; }
        public int GenericTypeParams { get; internal set; }
        public IList<String> Annotations { get; internal set; }
        public IList<MethodDescription> MethodDescriptions { get; internal set; }
        public IList<FieldDescription> FieldDescriptions { get; internal set; }

        #endregion

    }
}

File: ambeth-integrity/com.koch.classbrowser.java/src/main/java/com/koch/classbrowser/java/TypeDescription.java
package com.koch.classbrowser.java;

import java.util.ArrayList;
import java.util.List;

import org.apache.commons.lang3.StringUtils;

/**
 * Holds the description for types.
 *
 * @author juergen.panser
 */
public class TypeDescription implements INamed, IDeprecation {
	// ---- VARIABLES ----------------------------------------------------------

	private String source;

	private String moduleName;

	private String namespaceName;

	private String typeName;

	private String fullTypeName;

	private String typeType;

	private int genericTypeParams;

	private String superType;

	private final List<String> interfaces = new ArrayList<>();

	private final List<AnnotationInfo> annotations = new ArrayList<>();

	private final List<MethodDescription> methodDescriptions = new ArrayList<>();

	private final List<FieldDescription> fieldDescriptions = new ArrayList<>();

	// ---- CONSTRUCTORS -------------------------------------------------------

	/**
	 * Create a new instance.
	 *
	 * @param source Source; mandatory
	 * @param moduleName Module name; mandatory
	 * @param namespaceName Namespace name; optional
	 * @param typeName Short name of the type; mandatory (because no anonymous classes are allowed
	 *        where the canonical name may be null or the simple name be empty)
	 * @param fullTypeName Full name of the type including the namespace; mandatory
	 * @param typeType Type constant; mandatory
	 * @param genericTypeParams Number of generic type parameters
	 */
	public TypeDescription(String source, String moduleName, String namespaceName, String typeName,
			String fullTypeName, String typeType, int genericTypeParams) {
		if (StringUtils.isBlank(source) || StringUtils.isBlank(typeName)
				|| StringUtils.isBlank(fullTypeName) || StringUtils.isBlank(typeType)) {
			throw new IllegalArgumentException("Mandatory type description value missing!");
		}
		this.source = source;
		this.moduleName = moduleName;
		this.namespaceName = namespaceName;
		this.typeName = typeName;
		this.fullTypeName = fullTypeName;
		this.typeType = typeType;
		this.genericTypeParams = genericTypeParams;
	}

	// ---- GETTER/SETTER METHODS ----------------------------------------------

	public String getSource() {
		return source;
	}

	public String getModuleName() {
		return moduleName;
	}

	public String getNamespaceName() {
		return namespaceName;
	}

	@Override
	public String getName() {
		return typeName;
	}

	public String getFullTypeName() {
		return fullTypeName;
	}

	public String getTypeType() {
		return typeType;
	}

	public int getGenericTypeParams() {
		return genericTypeParams;
	}

	public String getSuperType() {
		return superType;
	}

	public void setSuperType(String superType) {
		this.superType = superType;
	}

	@Override
	public boolean isDeprecated() {
		return IDeprecation.INSTANCE.isDeprecated(annotations);
	}

	public List<String> getInterfaces() {
		return interfaces;
	}

	public List<AnnotationInfo> getAnnotations() {
		return annotations;
	}

	public List<MethodDescription> getMethodDescriptions() {
		return methodDescriptions;
	}

	public List<FieldDescription> getFieldDescriptions() {
		return fieldDescriptions;
	}
}
